### PR TITLE
Use golang-docker to create docker-test

### PR DESCRIPTION
**Pull request by ineiti**
Under MacOSX, the docker-test image doesn't contain the bevm contract, which makes it
impossible to run all tests in external/js/cothority.
This PR uses golang-docker to create the docker-test image, so that also under MacOSX
all tests from external/js/cothority can be run.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,9 +27,13 @@ proto:
 	./proto.sh
 	make -C external
 
+# 	docker run -t -v $(PWD):/cothority neilotoole/xcgo \
```
Comment (tharvik) [inline Makefile:30]: Dangling comment?

File: external/docker/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,8 @@
-FROM dedis/conode:dev
+FROM golang:1.15-buster
+RUN apt update && apt install -y procps ca-certificates && apt clean
+WORKDIR /root/
+
+EXPOSE 7770 7771
```
Comment (tharvik) [inline external/docker/Dockerfile:5]: It isn't in sync with what `CMD` starts, there is more openned ports.

**Review by tharvik — COMMENTED** [review]
It might be more readable to use a docker builder image à la `conode/Dockerfile`.

**Comment by ineiti** [comment]
I didn't find a way to build a docker-image with a `Dockerfile` while 'mounting' the local filesystem. Only `COPY` is available in the `Dockerfile`, which might be quite heavy...

**Comment by tharvik** [comment]
> I didn't find a way to build a docker-image with a `Dockerfile` while 'mounting' the local filesystem. Only `COPY` is available in the `Dockerfile`, which might be quite heavy...

Yep, using a local builder might create big temporary images, but if that's an issue, you can use the `.dockerignore` files.
I've recently done a PR on spindle using bother a builder image and ignoring some files: https://github.com/ldsec/spindle/pull/18

File: Makefile (hunk 1 above)
Comment (ineiti) [reply Makefile]: fixed

File: external/docker/Dockerfile (hunk 2 above)
Comment (ineiti) [reply external/docker/Dockerfile]: fixed

**Comment by ineiti** [comment]
Is there no way to copy only the "*.go" files to the docker-builder?

**Comment by tharvik** [comment]
> Is there no way to copy only the "*.go" files to the docker-builder?

Try something like the following `.dockerignore`
```
*
!*.go
```

**Comment by ineiti** [comment]
Anyway - thinking this through, to make it nice, I think a lot more would need to be changed. Just adding `Dockerfile` and `.dockerignore` to the root to get two lines out of the `Makefile` doesn't make it nicer.

So if you're OK, I'll leave it like this with the `docker-go` build command in the `Makefile`.

**Review by tharvik — APPROVED** [review]

**Pull request merged**